src/h3lib/lib/vec2d.c

```diff
@@ -55,12 +56,12 @@ void _v2dIntersect(const Vec2d *p0, const Vec2d *p1, const Vec2d *p2,
 }
 
 /**
- * Whether two 2D vectors are equal. Does not consider possible false
- * negatives due to floating-point errors.
+ * Whether two 2D vectors are almost equal, within some threshold
  * @param v1 First vector to compare
  * @param v2 Second vector to compare
- * @return Whether the vectors are equal
+ * @return Whether the vectors are almost equal
  */
-bool _v2dEquals(const Vec2d *v1, const Vec2d *v2) {
-    return v1->x == v2->x && v1->y == v2->y;
+bool _v2dAlmostEquals(const Vec2d *v1, const Vec2d *v2) {
+    return fabs(v1->x - v2->x) < FLT_EPSILON &&
+           fabs(v1->y - v2->y) < FLT_EPSILON;
 }
```

Comment:
Why does this function do comparisons in single precision float instead of double?

Reply:
`fabs` takes and returns a double, so the comparison here is still in double precision I believe. `FLT_EPSILON` was chosen as a reasonable threshold here; a number of boundary tests fail when we use the stricter `DBL_EPSILON` threshold.